### PR TITLE
Cospad refactor

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

Updating to match the template, and use software PWM backlight.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
The TKG firmware for the Cospad appears to be the same as that for the "GHPad", so the LED mappings may be the same: https://github.com/kairyu/tmk_keyboard_custom/blob/master/keyboard/ghpad/ledmap.c

It would be good if I could get someone with one of these to trace out the top four LEDs just to be sure - I'm a little confused that it says the "PWM" LED is on B5, but the QMK `BACKLIGHT_PIN` is F7...

File: keyboards/cospad/cospad.h [hunk 1]
```diff
@@ -14,12 +13,14 @@
...
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+
 #pragma once
 
 #include "quantum.h"
-#define ___ KC_NO
 
-/* This a shortcut to help you visually see your layout.
+#define XXX KC_NO
```
Comment (zvecr) [inline keyboards/cospad/cospad.h]: There are plenty of boards which use the `___` style, is there any reason to change this?

Comment (fauxpark) [reply keyboards/cospad/cospad.h]: I'd rather keep these 3-character versions consistent with the 7-character versions.

File: keyboards/cospad/cospad.c [hunk 2]
```diff
@@ -16,18 +15,9 @@
...
 
-#ifdef BACKLIGHT_ENABLE
-
-void backlight_init_ports(void) {
-  setPinOutput(F7);
-}
-
-void backlight_set(uint8_t level) {
-  writePin(F7, !!level);
+bool led_update_kb(led_t led_state) {
```
Comment (zvecr) [inline keyboards/cospad/cospad.c]: Tested on my board, doesnt work without...
```suggestion
void keyboard_pre_init_kb(void) {
    setPinOutput(B2);

    keyboard_pre_init_user();
}

bool led_update_kb(led_t led_state) {
```

File: keyboards/cospad/cospad.h (hunk 1 above)
Comment (zvecr) [reply keyboards/cospad/cospad.h]: There are 49 instances of `___` vs 64 for `xxx` or `XXX` and 9 for `_x_`.

While I agree with the case of consistency, I personally find `___` easier to read within the context of the LAYOUT macros.

File: keyboards/cospad/cospad.c (hunk 2 above)
Comment (fauxpark) [reply keyboards/cospad/cospad.c]: I knew there was something I forgot...

**Comment by fauxpark** [comment]
Good point. Did the needful in config.h.

**Pull request merged**